src/afa/afa.cc

```diff
@@ -34,29 +36,531 @@ const std::string Mata::Afa::TYPE_AFA = "AFA";
 
 std::ostream& std::operator<<(std::ostream& os, const Mata::Afa::Trans& trans)
 { // {{{
-	std::string result = "(" + std::to_string(trans.src) + ", " + trans.formula + ")";
+	std::string result = "(" + std::to_string(trans.src) + ", " 
+                       + std::to_string(trans.symb) + ", " 
+                       + std::to_string(trans.dst) + ")";
 	return os << result;
 } // operator<<(ostream, Trans) }}}
 
-
+/** This function adds a new transition to the automaton. It changes a transition
+* relation. 
+* @brief adds a new transition
+* @param trans a given transition
+*/
 void Afa::add_trans(const Trans& trans)
 { // {{{
-	this->transitions.push_back(trans);
+    assert(!this->transRelation.empty());
+
+    // Performs a transition using given src state and symbol
+    // in context of transitions which have been already added to the automaton
+    // If there is no result, a new transition will be created
+    // If the corresponding transition already exists, given 'dst' will be added to
+    // the transition
+    auto nodesPtr = perform_trans(trans.src, trans.symb);
+    if(!nodesPtr->empty())
+    {
+        // TODO: It would be nice to remove redundant clauses (like (1 || (1 && 2)))
+        // The clause (1 && 2) could be deleted. Solution:
+        // create closed set -> add the new transition -> extract the antichain ->
+        // store it to the transition (???). Working solution:
+        //        
+        // auto cl = ClosedSet<State>(ClosedSet<State>::upward_closed, 0, max_state_used, *nodesPtr);
+        // cl.insert(trans.dst);
+        // *nodesPtr = cl.get_antichain();
+        //        
+        // Is it worth it?         
```

Comment:
Discuss this with others. We should decide on this before we merge the PR.

Reply:
Thanks for your suggestion, we intend to discuss it.